### ISSUE TITLE
Please remove dependencies on **rgdal**, **rgeos**, and/or **maptools**

### ISSUE THREAD

**Issue by rsbivand** [garretrc/ggvoronoi#17]
This package depends on (depends, imports or suggests) **raster** and one or more of the retiring packages **rgdal**, **rgeos** or **maptools** (https://r-spatial.org/r/2022/04/12/evolution.html, https://r-spatial.org/r/2022/12/14/evolution2.html). Since **raster** `3.6.3`, all use of external FOSS library functionality has been transferred to **terra**, making the retiring packages very likely redundant. It would help greatly if you could remove dependencies on the retiring packages as soon as possible.

**Comment by rsbivand** [comment]
You will be aware, for example from:
https://r-spatial.org/r/2022/04/12/evolution.html,
https://r-spatial.org/r/2022/12/14/evolution2.html,
https://r-spatial.org/r/2023/04/10/evolution3.html and
https://rsbivand.github.io/csds_jan23/bivand_csds_ssg_230117.pdf and 
perhaps view https://www.youtube.com/watch?v=TlpjIqTPMCA&list=PLzREt6r1NenmWEidssmLm-VO_YmAh4pq9&index=1 
that `rgdal`, `rgeos` and `maptools` will be retired this
year, in October 2023. 

The dependency on 'rgeos' found to be rgeos::gBuffer rgeos::intersect by `pkgapi`, line 95 in R/voronoi_polygon.R, needs urgent attention. My feeling is that `sp` classes were never a good fit, in that `fortify` always was fallible (if intersections lead to holes, they may not be properly encoded). But just to keep the package on CRAN after `rgeos` retires in October 2023, coercing to `vect` and replacing `rgeos` with `terra` calls to GEOS should be OK. `rgeos::gBuffer` is I think used just to try to make the polygons valid - there are other possibilities. The `geos` package could be used in place of `terra`, but not for `sp` objects.

**Comment by rsbivand** [comment]
@garretrc See also https://r-spatial.org/r/2023/05/15/evolution4.html. Please take urgent steps to resolve this issue. No positive changes appear in the repository. Less than three months remain to retain this package on CRAN.

**Comment by rsbivand** [comment]
@garretrc Less than three weeks remain to fix this.

**Comment by rsbivand** [comment]
@garretrc R spatial infrastructure packages maptools, rgdal and rgeos will be archived by CRAN on Monday, October 16, 2023. Your package does not pass CMD check when these packages are not available. Expect your package to be archived by CRAN October 17-18 as CRAN checks feed through and your package fails, if not updated by Monday, October 16, 2023. 

No grace period is anticipated, as you have had sufficient time to update your package to remove dependencies on maptools, rgdal and/or rgeos. It remains the case that many packages importing the raster package needlessly depend on retiring packages, as raster stopped using them a year ago.

**Comment by mhurk** [comment]
Cannot run my scripts which use ggvoronoi anymore. That's a pity and reading the above that was probably since October 16.

```ERROR: dependency 'rgeos' is not available for package 'ggvoronoi'```
```* removing 'C:/Users/<username>/AppData/Local/R/win-library/4.3/ggvoronoi'```

**Comment by rsbivand** [comment]
@mhurk @garretrc maybe consider taking over maintenance of the package? Try emailing the maintainers to get their approval. Please let me know if you need help.

**Comment by mhurk** [comment]
Thanks for the suggestion but tempting as it is I really don't have time at the moment to look into details of this or even try to fix it. But I will give it some thought and perhaps try to make some changes.

**Comment by rsbivand** [comment]
@mhurk Either convince @garretrc to merge my PR, and consider resubmitting to CRAN himself, or install from my fork https://github.com/rsbivand/ggvoronoi, possibly taking over maintenance by copying out my fork. The time involved is trivial, why the maintainer didn't do this themselves is unknown.

**Comment by mhurk** [comment]
@rsbivand Thank you for your quick fix. It works like a charm. Very helpful! And @garretrc thanks for the merge!

**Comment by garretrc** [comment]
Hasn't been a priority lately due to finishing phd. Thanks @rsbivand , getting 401 error when submitting to CRAN but will investigate on a different network later!

**Comment by rsbivand** [comment]
Yes, CRAN re-opens for submissions January 8: https://stat.ethz.ch/pipermail/r-package-devel/2023q4/010240.html.

**Comment by spittssj** [comment]
Is there any update on this @garretrc @rsbivand ? Could someone resubmit it to CRAN?

I'd like to use ggvoroni but would prefer to download from CRAN (and build a conda package) if possible.

**Comment by garretrc** [comment]
> Is there any update on this [@garretrc](https://github.com/garretrc) [@rsbivand](https://github.com/rsbivand) ? Could someone resubmit it to CRAN?
> 
> I'd like to use ggvoroni but would prefer to download from CRAN (and build a conda package) if possible.

Unfortunately this isn't the only issue preventing a CRAN release, I'll see if I can figure it out now that I'm done with grad school. For now I recommend installing via github. Since this package released, this sort of diagram is now easy to create using standard R spatial packages, see https://www.jla-data.net/eng/spatial-aggregation/. I plan to include a code comparison in the readme since this is likely not the best solution for certain users anymore.

**Comment by spittssj** [comment]
I appreciate the link. It's helpful as I learn about my options.

I found out that conda-skeleton can build packages directly from Github as well, so I was able to build the package this afternoon and I'll test it this weekend. I find that the r spatial ecosystem is a little fragile, so I like to use conda environments when possible.

I'm migrating some Voroni diagrams I built manually in ArcGIS a while back to a reproducible research framework now that I have an update of the underlying data set.

Congrats on a PhD from a fellow Big 10 school. I'm on the job market this year from UMinnesota.
